src/main/java/com/aws/greengrass/clientdevices/auth/iot/IotClientFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot;
+
+import com.aws.greengrass.deployment.DeviceConfiguration;
+import com.aws.greengrass.deployment.exceptions.DeviceConfigurationException;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.IotSdkClientFactory;
+import com.aws.greengrass.util.Utils;
+import software.amazon.awssdk.services.iot.IotClient;
+
+import javax.inject.Inject;
+import java.net.URISyntaxException;
+
+public class IotClientFactory {
+
+    private final DeviceConfiguration deviceConfiguration;
+
+    @Inject
+    public IotClientFactory(DeviceConfiguration deviceConfiguration) {
+        this.deviceConfiguration = deviceConfiguration;
+    }
+
+    public IotClient getClient() throws DeviceConfigurationException {
+        try {
+            return IotSdkClientFactory.getIotClient(
+                    getAwsRegion(deviceConfiguration),
+                    IotSdkClientFactory.EnvironmentStage.PROD // TODO
+            );
+        } catch (URISyntaxException unused) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While re-throwing the caught exception with modifications , information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)

Suggested remediation:
Consider passing the exception object either as  &#34;rootCause&#34; or as an inner exception parameter, to the constructor of the new exception before throwing new exception. This will help keep stack trace.
```
@@ -34,3 +34,3 @@
             // this will never execute. exception can only throw when stage != EnvironmentStage.PROD
-            throw new DeviceConfigurationException(&#34;&#34;);
+            throw new DeviceConfigurationException(&#34;&#34;, unused);
         }

```